indico/modules/events/contributions/controllers/display.py

```diff
@@ -53,8 +55,11 @@ class RHContributionDisplayBase(RHDisplayEventBase):
 
     def _check_access(self):
         RHDisplayEventBase._check_access(self)
-        if not self.contrib.can_access(session.user):
+        published = contribution_settings.get(self.event, 'published')
+        if not published or not self.contrib.can_access(session.user):
             raise Forbidden
+        if not published:
```

Comment:
this code never runs since you raise Forbidden above...

Reply:
I should just remove the `not published` check from there. Probably a mistake.